python/pynessie/client/nessie_client.py

```diff
@@ -259,7 +263,10 @@ def merge(self, from_ref: str, onto_branch: str, from_hash: Optional[str] = None
             from_hash = from_hash_ref
 
         merge_json = MergeSchema().dump(Merge(from_ref, str(from_hash)))
-        merge(self._base_url, self._auth, onto_branch, merge_json, old_hash, self._ssl_verify)
+        merge_response = merge(self._base_url, self._auth, onto_branch, merge_json, old_hash, self._ssl_verify)
+        if merge_response:
+            return MergeResponseSchema().load(merge_response)
+        return None
```

Comment:
we should describe with a comment and/or in the method docs when `None` can be returned i.e. we only expect this to happen against legacy servers... starting from a specific server version it would be an error to get an empty response

Reply:
Added to method docs.